### PR TITLE
Login: fix the switch to suggested language and log-in redirection

### LINKED ISSUE #61358: [Bug]: locale switcher does not change page content
### Quick summary

The locale switcher banner -- the one that says `Also available in English` or equivalent -- changes the URL but not the page content. I am guessing a reload is needed, but either it does not perform one or it fails while doing so.

### Steps to reproduce

1. While logged out (e.g. in a fresh incognito window), navigate to `https://wordpress.com/log-in/<locale>`, where `<locale>` is some locale different from your browser settings. e.g. https://wordpress.com/log-in/fr (French)
2. You should get the WordPress.com login page, in the locale chosen, with a switcher banner at the top of the form. If not, maybe try another locale.
3. Click the locale switcher.
4. Notice the page did not change.
5. If you check the URL, the locale slug got switched.

### What you expected to happen

I expected the page contents to update to my chosen language when I clicked the switcher.

### What actually happened

The URL got updated, but no reload happened, and no page change happened.

### Context

Discovered while accepting a user invite while logged out

### Browser

Google Chrome/Chromium, Apple Safari

### Simple/Atomic

Simple

### Other notes

This also applies to other paths where the locale switcher pops up, e.g. `https://wordpress.com/accept-invite/<blog-id>/<invite-key>/<locale>` (while logged out).

No browser console errors observed.

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Workaround details

Since the URL is already correct, the user can manually load this to get the correctly translated page.

### PR DESCRIPTION
**Pull request by sejas**
#### Changes proposed in this Pull Request

- Simplify the login Routes middleware to fix the translation in Login
- Fix the login redirection when the user clicks on the LocalSuggestion to English. From `/log-in/en` to `/log-in`

#### Screenshots

**Before**

https://user-images.githubusercontent.com/779993/166646033-b927b1ed-b407-4721-b054-dfb438602e7d.mov

**After**

https://user-images.githubusercontent.com/779993/166646040-dbd2d0be-5ec0-48bb-8d87-729de0cc7ead.mov

#### Testing instructions

1. Your browser should have English as your preferred language.
2. Go to the login screen with any locale different from `en` : [`/log-in/fr`](https://wordpress.com/log-in/fr) 
3. Observe the page is rendered in "french"
4. Click on the Locale Suggestion. It contains this text: `Also available in English`
5. Observe that now the page renders correctly in English and the URL has removed the `/en` from the URL.
6. The switch to the LocalSuggestion component is in multiple places, but I've found that the issue only happened on the log-in screen.
7. Repeat from step 2 in the jetpack log-in URL : [`/log-in/jetpack/fr`](https://wordpress.com/log-in/jetpack/fr)

- Closes #61358

### CODE REVIEW

**Review by mmtr — APPROVED** [review]
This tests well for me, fantastic job! However, I'm not fully confidence whether these changes are safe, since they touch a critical area of Calypso (the user authentication! 😱 ), so I'd appreciate if someone else could take a look at this.

Maybe @Tug could help since he worked on this same `redirectDefaultLocale` controller in https://github.com/Automattic/wp-calypso/pull/19319?

**Review by p-jackson — COMMENTED** [review]
I gave this a test and found that the only thing that needed to change in order to fix #61358 was changing the order of the route handler functions. The simplification of the `redirectDefaultLocale()` function doesn't appear to be needed, and I'm worried it removes important behaviour.

I'm not sure what the `wpcom-user-bootstrap` feature is (hopefully @Tug can shed some light). But the code that's used to prevent redirects when the user is already logged in is important.

When you visit `/log-in` while logged in you see this screen
Screenshot 2022-05-05 at 11 20 03 AM

And since the user is logged in, we have a better idea of what language to show without referring to the browser preference: we use the language that's set in `/me/account`. That's why `redirectDefaultLocale()` doesn't remove the `/en` suffix if the user is a non-en. If their account is set to French and the URL has explicitly set the suffix to `/en` then they're trying to force the UI to show English and stripping the suffix will revert it back to French.

Now in saying all that, there appears to be another bug in play. The language suffix feature doesn't seem to be working when the user is logged in. While logged in the `/log-in/ja` and `/log-in/es` URLs render the language using the account setting language. This isn't how the `setLocaleMiddleware` handler is supposed to work 🤔. This is a different bug from #61358 though so should perhaps be dealt with seperately.

File: client/login/controller.js [hunk 1]
```diff
@@ -122,31 +121,12 @@ export function magicLoginUse( context, next ) {
...
-	}
-
-	if ( context.params.isJetpack === 'jetpack' ) {
+	if ( context.pathname === '/log-in/jetpack/en' ) {
 		context.redirect( '/log-in/jetpack' );
-	} else {
+	} else if ( context.pathname === '/log-in/en' ) {
 		context.redirect( '/log-in' );
 	}
+	next();
```
Comment (Tug) [inline client/login/controller.js]: Looks wrong that we redirect but still call the next middlewares

**Review by Tug — COMMENTED** [review]
It's been more than 4 years I've worked on this so I couldn't really tell. I can give it a quick test.

**Review by Tug — APPROVED** [review]
Looks good, I didn't see any issue while testing 👍

**Review by mmtr — APPROVED** [review]
Thanks for the latest changes @sejas, they are much more targeted and safer now. I re-tested them and everything worked as expected. 

@p-jackson I'd appreciate if you can give this a final review as well before merging, thanks!

**Comment by sejas** [comment]
Thank you @mmtr !!

I've decided to keep the user bootstrap code in place.
And also, keep the  @p-jackson mentioned before when the user is logged, and his language preference is different from `en`.

@p-jackson let me know what you think.

A few interesting findings are:
- `context.redirect` is a custom function that calls to `page.redirect`and it replaces the `url`, finishing the middleware flow at that point.
- `page.redirect` re-starts the route middleware from the beginning, but it removes all the current context. That's why I need to set the `en` in params, to force changing the current language.

File: client/login/controller.js [hunk 2]
```diff
@@ -131,6 +131,9 @@ export function magicLoginUse( context, next ) {
 export function redirectDefaultLocale( context, next ) {
 	// Only handle simple routes
 	if ( context.pathname !== '/log-in/en' && context.pathname !== '/log-in/jetpack/en' ) {
+		if ( ! isUserLoggedIn( context.store.getState() ) && ! context.params.lang ) {
+			context.params.lang = 'en';
```
Comment (mmtr) [inline client/login/controller.js]: If the intent here is to load the default locale, we may prefer using `getLocaleSlug` instead:

https://github.com/Automattic/wp-calypso/blob/f39dd4c373389bc87d227a3a022f2eb386e29e9c/client/lib/i18n-utils/node.js#L2

Comment (mmtr) [reply client/login/controller.js]: We can also run `config( 'i18n_default_locale_slug' )` directly, as done in other places:

https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/notifications/index.jsx#L171

Comment (sejas) [reply client/login/controller.js]: Thanks for the suggestion.
I've updated it. It looks better using the default lang from the config function.

Thank you!

**Review by p-jackson — APPROVED** [review]
Nice find with that difference between `context.redirect` and `page.redirect`.

Clicking the suggested language link is now always changing the UI instantly, as expected 👍 

I am still seeing issues in the already-logged-in case where the URL param can't be used to set the display language. In the below example my user preference is French, my browser preference is English, and I've tried to force the UI to Japanese using the URL.
Screenshot 2022-05-12 at 2 29 21 PM

But it also behaves this way in production so this PR hasn't introduced the bug. It's probably not a common use case either, just indicative that the controller isn't obeying the URL very well. At least clicking "English" in this case does update the UI now 😄 

Anyway, ignore my commentary about this separate bug. Nice work getting the language switcher fixed, this is good to go 🚢

**Pull request merged**